zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -145,11 +146,13 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
             0.expr(),
             None,
         );
+        let is_caller_callee_equal = cb.query_bool();
         cb.account_access_list_write(
             tx_id.expr(),
             tx_callee_address.expr(),
             1.expr(),
-            0.expr(),
+            // Freely choosen by prover, not a soundness issue
+            is_caller_callee_equal.expr(),
```

Comment:
Strictly speaking it's not that freely chosen by prover, the prover still needs to pass the State circuit consistency check, so we don't need extra constraint here.

If `caller == callee`, then the consistency check will enforce `is_caller_callee_equal == 1`.

If `caller != callee`, then the State circuit initial value's rule will enforce `is_caller_callee_equal == 0`.

https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/30a2b10f57e508948fb12b3a7c3f63fdeb174df2/zkevm-circuits/src/state_circuit/constraint_builder.rs#L315-L318

Reply:
I will update the comment 👍